### PR TITLE
[Front] Favorites preview

### LINKED ISSUE #7707: [Favorites] UX: change assistant modal menu to button bar
See figma below

Link to [Figma](https://www.figma.com/design/QOxHwppG64GtPocpDhS6Y7/Product?node-id=7033-45013&t=gUYgBFDqdJbosTa1-4)
Link to [Notion page](https://www.notion.so/dust-tt/Assistants-Favorites-50277abe540b4d5480f2f78e119c31fb?pvs=4)

### LINKED ISSUE #7708: [Favorites] UX: Rename "in my list" tab to "favorite" and make it appear first if not empty
Link to [Figma](https://www.figma.com/design/QOxHwppG64GtPocpDhS6Y7/Product?node-id=7033-45013&t=gUYgBFDqdJbosTa1-4)
Link to [Notion page](https://www.notion.so/dust-tt/Assistants-Favorites-50277abe540b4d5480f2f78e119c31fb?pvs=4)

### LINKED ISSUE #7709: [Favorites] UX: favorites appear first on @ mention (and in dropdown list)
Link to [Figma](https://www.figma.com/design/QOxHwppG64GtPocpDhS6Y7/Product?node-id=7033-45013&t=gUYgBFDqdJbosTa1-4)
Link to [Notion page](https://www.notion.so/dust-tt/Assistants-Favorites-50277abe540b4d5480f2f78e119c31fb?pvs=4)

### LINKED ISSUE #7710: [Favorites] UX: favorites appear first on suggestions if no agent is mentioned
Link to [Figma](https://www.figma.com/design/QOxHwppG64GtPocpDhS6Y7/Product?node-id=7033-45013&t=gUYgBFDqdJbosTa1-4)
Link to [Notion page](https://www.notion.so/dust-tt/Assistants-Favorites-50277abe540b4d5480f2f78e119c31fb?pvs=4)

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- Close #7710
- Close #7708 
- Close #7709 

**Note**: the changes in this CL rely on the old "in-list" parameters and are not plugged on changes in the backend that are related to the new Favorites feature.

## Deploy Plan

- Deploy front

### CODE REVIEW

File: front/components/assistant/conversation/input_bar/editor/suggestion.ts [hunk 1]
```diff
@@ -39,6 +39,11 @@ export function makeGetAssistantSuggestions() {
       const { suggestions, fallbackSuggestions } =
         editor.storage.MentionStorage.suggestions;
 
+      /// keeping the pre-defined order when queried without content
```
Comment (philipperolet) [inline front/components/assistant/conversation/input_bar/editor/suggestion.ts:42]: is that in the scope of this PR? why introduce that?

**Review by philipperolet — APPROVED** [review]
Thanks, looks great :+1: Left a com but can approve right now :)

**Comment by aubin-tchoi** [comment]
Suggestions

Screenshot 2024-10-11 at 5 45 39 PM

Mentions

Screenshot 2024-10-11 at 5 45 33 PM

Tabs

Screenshot 2024-10-11 at 5 45 26 PM

File: front/components/assistant/conversation/input_bar/editor/suggestion.ts (hunk 1 above)
Comment (aubin-tchoi) [reply front/components/assistant/conversation/input_bar/editor/suggestion.ts:42]: Yes, this is specifically the change that sorts the mentions by overriding the usual fuzzy/spreadLength search.

Comment (aubin-tchoi) [reply front/components/assistant/conversation/input_bar/editor/suggestion.ts:42]: I hope I didnt misunderstand #7709

**Pull request merged**